File: src/backend/Hermes.SituationRoom.Data/Entities/PrivacyLevelPersonal.cs
using System;
using System.Collections.Generic;

namespace Hermes.SituationRoom.Data.Entities;

public partial class PrivacyLevelPersonal
{
    public Guid Uid { get; set; }

    public bool IsFirstNameVisible { get; set; }

    public bool IsLastNameVisible { get; set; }

    public bool IsEmailVisible { get; set; }

    public Guid OwnerUid { get; set; }

    public Guid ConsumerUid { get; set; }

    public virtual User ConsumerU { get; set; }

    public virtual User OwnerU { get; set; }
}

File: src/frontend/src/types/privacy-level-personal.ts
export interface PrivacyLevelPersonalBo {
    uid: string;
    isFirstNameVisible: boolean;
    isLastNameVisible: boolean;
    isEmailVisible: boolean;
    ownerUid: string;
    consumerUid: string;
}

export interface CreatePrivacyLevelDto {
    isFirstNameVisible: boolean;
    isLastNameVisible: boolean;
    isEmailVisible: boolean;
    ownerUid: string;
    consumerUid: string;
}

export interface UpdatePrivacyLevelDto {
    isFirstNameVisible: boolean;
    isLastNameVisible: boolean;
    isEmailVisible: boolean;
}